controllers/utils/utils.go

```diff
@@ -133,3 +136,91 @@ func GetManagementOwnerRef(capiMachine *capiv1.Machine) *meta.OwnerReference {
 	}
 	return fetchOwnerRef(capiMachine.OwnerReferences, "EtcdadmCluster")
 }
+
+// GetOwnerOfKind returns the Cluster object owning the current resource of passed kind.
+func GetOwnerOfKind(ctx context.Context, c clientPkg.Client, owned client.Object, owner client.Object) error {
+	gvks, _, err := c.Scheme().ObjectKinds(owner)
+	if err != nil {
+		return errors.Wrapf(err, "error encountered when finding owner kind for %s/%s", owned.GetName(), owned.GetNamespace())
+	} else if len(gvks) != 1 {
+		return errors.Errorf(
+			"found more than one GVK for owner when finding owner kind for %s/%s", owned.GetName(), owned.GetNamespace())
+	}
+	kind := gvks[0].Kind
+	for _, ref := range owned.GetOwnerReferences() {
+		if ref.Kind != kind {
+			continue
+		}
+		key := client.ObjectKey{Name: ref.Name, Namespace: owned.GetNamespace()}
+		if err := c.Get(ctx, key, owner); err != nil {
+			return errors.Wrapf(err, "error encountered when finding owner of kind %s %s/%s",
+				owner.GetObjectKind().GroupVersionKind().Kind, owner.GetNamespace(), owner.GetName())
+		}
+		return nil
+	}
+	return errors.Errorf("couldn't find owner of kind %s %s/%s",
+		owner.GetObjectKind().GroupVersionKind().Kind, owner.GetNamespace(), owner.GetName())
+}
+
+// GetOwnerCloudStackCluster returns the Cluster object owning the current resource.
+func GetOwnerCloudStackCluster(ctx context.Context, c clientPkg.Client, obj metav1.ObjectMeta) (*capcv1.CloudStackCluster, error) {
+
+	for _, ref := range obj.OwnerReferences {
+		if ref.Kind != "CloudStackCluster" {
+			continue
+		}
+		gv, err := schema.ParseGroupVersion(ref.APIVersion)
+		if err != nil {
+			return nil, errors.WithStack(err)
+		}
+		if gv.Group == capcv1.GroupVersion.Group {
+			csCluster := &capcv1.CloudStackCluster{}
+			key := client.ObjectKey{Name: ref.Name, Namespace: obj.Namespace}
+			if err := c.Get(ctx, key, csCluster); err != nil {
+				return nil, err
+			}
+			return csCluster, nil
+		}
+	}
+	return nil, nil
+}
+
+// GetOwnerZone returns the Cluster object owning the current resource.
```

Comment:
Is this comment accurate?

Reply:
I'm also not sure about its code.  Is it looking up a Cluster, not a Zone, and slamming the result into a Zone struct?

I think this func is unused.

Reply:
Yup, removing. This method and the cluster equivalent are completely unneeded not that I managed to put together a getParent generic version.